pandas/core/groupby.py

```diff
@@ -4585,9 +4632,18 @@ def _apply_to_column_groupbys(self, func):
              in self._iterate_column_groupbys()),
             keys=self._selected_obj.columns, axis=1)
 
+    def _fill(self, direction, limit=None):
+        """Overriden method to concat grouped columns in output"""
+        res = super()._fill(direction, limit=limit)
+        output = collections.OrderedDict()
```

Comment:
would be slightly simpler as a list-comprehension

Reply:
though *maybe* this should be an arg to _fill? (IOW the option to do it should be done in fill)

Reply:
Do you mean as an arg to `_get_cythonized_result` (fill would always use, unless we wanted to implement a new feature)? I was thinking about that but given `fill` on a `DataFrameGroupBy` is the only operation I know of at the moment that includes the grouping in the body of the returned object I figured it made the most sense to just perform that action there